core/src/components/modal/gestures/swipe-to-close.ts

```diff
@@ -39,16 +39,18 @@ export const createSwipeToCloseGesture = (
   };
 
   const onMove = (detail: GestureDetail) => {
-    const step = detail.deltaY / height;
-    if (step < 0) { return; }
+    const step = getStep(detail.deltaY, height);
 
     animation.progressStep(step);
   };
 
   const onEnd = (detail: GestureDetail) => {
-    const velocity = detail.velocityY;
-    const step = detail.deltaY / height;
-    if (step < 0) { return; }
+    let velocity = detail.velocityY;
+    if (velocity <= 0) {
+      velocity = 0.0001; // Fixes modal stuck when swiping up from the bottom very fast
+    }
+
+    const step = getStep(detail.deltaY, height);
```

Comment:
We can just use the `clamp` function here instead:
```suggestion
    const velocity = detail.velocityY;
    const step = clamp(0.0001, detail.deltaY / height, 0.9999);
```

This will ensure that the "ending" values are always in-bounds